datasetinsights/io/downloader/unity_simulation.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import os
+import re
+
+from datasetinsights.io.downloader.base import DatasetDownloader
+from datasetinsights.io.usim import Downloader, download_manifest
+
+logger = logging.getLogger(__name__)
+
+
+class UnitySimulationDownloader(DatasetDownloader, protocol="usim://"):
+    """ This class is used to download data from Unity Simulation
+
+        For more on Unity Simulation please see these
+        `docs <https://github.com/Unity-Technologies/Unity-Simulation-Docs>`
+
+        Args:
+        access_token: "Access token to be used to authenticate to
+         unity simulation for downloading the dataset"
+
+    """
+
+    SOURCE_URI_PATTERN = (
+        r"usim://([^@]*)?@?"
+        r"([a-fA-F0-9]{8}-"
```

Comment:
[nit] might be OK to have UUID regex in a single line (improve readability). If this violates pep8, you can add `# noqa: E501` to skip flake8 error for this line.

Reply:
will do